nautobot/core/tests/test_templatetags_ui_framework.py

```diff
@@ -0,0 +1,146 @@
+from django.template import Context
+
+from nautobot.core.templatetags import ui_framework
+from nautobot.core.testing import TestCase
+from nautobot.core.ui.breadcrumbs import Breadcrumbs
+from nautobot.core.ui.titles import Titles
+
+
+class NautobotTemplatetagsUIComponentsTest(TestCase):
+    """Tests template tags from ui_framework module."""
+
+    # ---------------------------
+    # render_title
+    # ---------------------------
+
+    def test_render_title_with_legacy_title_present(self):
+        context = Context(
+            {
+                "title": "Custom Title",
+                "view_titles": Titles(),
+                "verbose_name_plural": "Default Title",
+            }
+        )
+        output = ui_framework.render_title(context)
+
+        self.assertEqual(output, "Custom Title")
+
+    def test_render_title_with_view_titles_only(self):
+        context = Context(
+            {
+                "view_titles": Titles(),
+                "verbose_name_plural": "Default Title",
+            }
+        )
+        output = ui_framework.render_title(context)
+
+        self.assertEqual(output, "Default Title")
+
+    def test_render_title_with_invalid_view_titles(self):
+        class MyStuff:
+            pass
+
+        context = Context(
+            {
+                "view_titles": MyStuff(),
+                "verbose_name_plural": "Default Title",
+            }
+        )
+        output = ui_framework.render_title(context)
+
+        self.assertEqual(output, "")
+
+    def test_render_title_with_empty_context(self):
+        context = Context({})
+        output = ui_framework.render_title(context)
+
+        self.assertEqual(output, "")
+
+    # ---------------------------
+    # render_breadcrumbs
+    # ---------------------------
+
+    def test_render_breadcrumbs(self):
+        context = Context(
+            {
+                "list_url": "home",
+                "title": "New Home",
+                "view_action": "list",
+                "breadcrumbs": Breadcrumbs(),
```

Comment:
Similarly, any test needed for a non-default `Breadcrumbs` object, or is that also covered in other tests already?